examples/medplum-chart-demo/src/bots/core/gynecology-encounter-note.ts

```diff
@@ -0,0 +1,395 @@
+  }
+
+  const conditionData: ConditionData = {
+    reasonForVisit: answers['reason-for-visit'].valueCoding,
+    problemList: answers['problem-list']?.valueBoolean ?? false,
+  };
+
+  const clinicalImpressionData: ClinicalImpressionData = {
+    visitLength: answers['visit-length']?.valueInteger,
+    assessment: answers['assessment']?.valueString,
+  };
+
+  // Create bundle entries from the above objects
+  const observationEntries = createObservationEntries(observationData, encounter, user);
+  const conditionEntry = createConditionEntry(conditionData, encounter, user);
+  const clinicalImpressionEntry = createClinicalImpressionEntry(clinicalImpressionData, encounter, user);
+
+  // Create an array of bundle entries for all resource types
+  const entry = [...observationEntries, ...conditionEntry];
+  if (clinicalImpressionEntry) {
+    entry.push(clinicalImpressionEntry);
+  }
+
+  // Create a batch
+  const resourceBatch: Bundle = {
+    resourceType: 'Bundle',
+    type: 'batch',
+    entry,
+  };
+
+  // Execute the batch to create all resources at once
+  const responseBundle = await medplum.executeBatch(resourceBatch);
+  return responseBundle;
+}
+
+function createObservationEntries(
+  observationData: ObservationData,
+  encounter: Encounter,
+  user: Practitioner
+): BundleEntry[] {
+  const entries: BundleEntry[] = [];
+  const request: BundleEntryRequest = { method: 'POST', url: 'Observation' };
+  // Create a generic observation that just needs to have a code and value added to it
+  const genericObservation: Observation = {
+    resourceType: 'Observation',
+    status: 'preliminary',
+    subject: encounter.subject,
+    encounter: { reference: getReferenceString(encounter) },
+    performer: [{ reference: getReferenceString(user) }],
+    effectiveDateTime: observationData.date,
+    code: {},
+  };
+
+  // Loop over each entry of the observation data
+  for (const [key, value] of Object.entries(observationData)) {
+    // If there is no value for a key, skip
+    // Check that blood pressure has a measurement for at least one of systolic and diastolic, otherwise skip
+    // Date is used to get the time for each observation, but does not create its own observation
+    if (!value || (key === 'bloodPressure' && !value.systolic && !value.diastolic) || key === 'date') {
+      continue;
+    }
+
+    // Create an entry and add it to the entries array
+    entries.push(createObservationEntry(key, request, genericObservation, observationData));
+  }
+
+  return entries;
+}
+
+function createObservationEntry(
+  key: string,
+  request: BundleEntryRequest,
+  genericObservation: Observation,
+  observationData: ObservationData
+): BundleEntry {
+  // Create a base resource from the generic
+  const resource: Observation = {
+    ...genericObservation,
+    code: {},
+  };
+
+  // Add the appropriate code and value based on the key
+  switch (key) {
+    case 'lastPeriod':
```

Comment:
Can you make this a map from `linkId` -> `Coding`. Place at the bottom of the file,